File: backend/schemas/logs.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class LogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int | None
    timestamp: datetime
    action: str
    table_cible: str
    item_id: int | None
    detail: str | None


class OcrStatRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    timestamp: datetime
    nom_fichier: str
    type_document: str
    type_mime: str
    taille_fichier: int
    nb_pages: int
    nb_champs_extraits: int
    nb_champs_vides: int
    taux_completude: float
    duree_ms: int
    duree_ocr_ms: int
    duree_extraction_ms: int
    succes: bool
    resultat_json: str | None

File: frontend/src/app/services/logs.ts
import api from "./api";

export interface LogEntry {
  id: number;
  user_id: number | null;
  timestamp: string;
  action: string;
  table_cible: string;
  item_id: number | null;
  detail: string | null;
}

export interface OcrStat {
  id: number;
  user_id: number;
  timestamp: string;
  nom_fichier: string;
  type_document: string;
  type_mime: string;
  taille_fichier: number;
  nb_pages: number;
  nb_champs_extraits: number;
  nb_champs_vides: number;
  taux_completude: number;
  duree_ms: number;
  duree_ocr_ms: number;
  duree_extraction_ms: number;
  succes: boolean;
  resultat_json: string | null;
}

export interface LogsParams {
  action?: string;
  table_cible?: string;
  user_id?: number;
  limit?: number;
  offset?: number;
}

export async function fetchLogs(params?: LogsParams): Promise<LogEntry[]> {
  const response = await api.get<LogEntry[]>("/logs/", { params });
  return response.data;
}

export async function fetchOcrStats(params?: { limit?: number; offset?: number }): Promise<OcrStat[]> {
  const response = await api.get<OcrStat[]>("/logs/ocr", { params });
  return response.data;
}

export async function restoreLog(logId: number): Promise<void> {
  await api.post(`/logs/${logId}/restore`);
}
